File: src/main/java/com/mini/credit/dto/credit/DecaissementCreditRequest.java
package com.mini.credit.dto.credit;

import com.mini.credit.enums.ModePaiement;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
public class DecaissementCreditRequest {

    @NotNull
    private LocalDateTime dateDecaissement;

    @NotNull
    private Long sessionCaisseId;

    private Long agentId;

    @NotNull
    private Long createdBy;

    @NotNull
    private ModePaiement modePaiement;

    private String observation;
}

File: frontend/src/app/features/credit/models/decaissement-credit-request.ts
import { ModePaiement } from '../../../shared/enums/mode-paiement.enum';

export interface DecaissementCreditRequest {
  dateDecaissement: string;
  sessionCaisseId: number;
  createdBy: number;
  modePaiement: ModePaiement;
  observation?: string | null;
}